slo-libsonnet/latency-burn.libsonnet

```diff
@@ -0,0 +1,109 @@
+local util = import '_util.libsonnet';
+{
+  burn(param):: {
+    local slo = {
+      metric: error 'must set metric for latency burn',
+      selectors: error 'must set selectors for latency burn',
+      latencyTarget: error 'must set latencyTarget latency burn',
+      latencyBudget: error 'must set latencyBudget latency burn',
+      labels: [],
+    } + param,
+
+    local rates = ['5m', '30m', '1h', '2h', '6h', '1d', '3d'],
+
+    local labels =
+      util.selectorsToLabels(slo.selectors),
+
+    local latencyRules = [
+      {
+        # How many percent are above the SLO latency target.
+        # First calculate how many requests are below the target and
+        # substract those from 100 percent.
+        # This gives the total requests that fail the SLO
+        expr: |||
+          1 - (
+            sum(rate(%s{%s,le="%s",code!~"5.."}[%s]))
+            /
+            sum(rate(%s{%s}[%s]))
+          )
+        ||| % [
+          slo.metric + "_bucket",
+          std.join(',', slo.selectors),
+          slo.latencyTarget,
+          rate,
+          slo.metric + "_count",
+          std.join(',', slo.selectors),
+          rate,
+        ],
+        record: 'latency%s:%s:rate%s' % [slo.latencyTarget+"s",slo.metric, rate],
```

Comment:
I'm thinking, rather than having this called `latency1s:` what do you think about simply naming it `latencytarget:`?

Reply:
wouldn't be better to know how much is the latency target by looking at the recording rule. Similiard to `rate%s`

Reply:
I'm fine with that too, just wanted to raise the point.